server_helen_comments.py

```diff
@@ -0,0 +1,446 @@
+show_blueprint = manager.create_api(Show, methods=['GET'])
+brand_blueprint = manager.create_api(Brand, methods=['GET'])
+color_blueprint = manager.create_api(Color, methods=['GET'])
+show_color_blueprint = manager.create_api(Show_Color, methods=['GET'])
+
+# Required to use Flask sessions and the debug toolbar
+app.secret_key = "ABC"
+
+# Normally, if you use an undefined variable in Jinja2, it fails silently.
+# This is horrible. Fix this so that, instead, it raises an error.
+app.jinja_env.undefined = StrictUndefined
+app.jinja_env.auto_reload = True
+
+
+@app.route('/')
+def index():
+    """Homepage."""
+    shows = Show.query.all()
+    show_colors = Show_Color.query.all()
+    colors = Color.query.all()
+    brands = Brand.query.all()
+
+    return render_template("bleep.html",
+                           shows=shows,
+                           show_colors=show_colors,
+                           colors=colors,
+                           brands=brands)
+
+
+# @app.route('/api/all')
+# def api_all():
+#     shows = Show.query.all()
+#     show_colors = Show_Color.query.all()
+#     colors = Color.query.all()
+#     brands = Brand.query.all()
+
+#     resp = Response(response=jsonify({'key': 'value'}),
+#                     status=200,
+#                     mimetype="application/json")
+#     return resp
+
+
+@app.route('/_get_brands')
+def get_brands_json():
+    brands = {}
+    for brand in Brand.query.all():
+        brands[brand.brand_id] = {
+            'brand_name': brand.brand_name,
+        }
+
+    return jsonify(brands)
+
+
+@app.route('/_get_shows')
+def get_shows_json():
+    shows = {}
+    for show in Show.query.all():
+        shows[show.show_id] = {
+            'show_id': show.show_id,
+            'show_season': show.season,
+            'show_year': show.year,
+            'brand_name': show.brands.brand_name,
+        }
+
+    return jsonify(shows)
+
+
+@app.route('/_get_colors')
+def get_colors_json():
+    colors = {}
+    for color in Color.query.all():
+        colors[color.color_id] = {
+            'color_id': color.color_id,
+            'color': color.color_name,
+            'color_hex': color.color_hex,
+        }
+
+    return jsonify(colors)
+
+
+@app.route('/_get_show_colors')
+def get_show_colors_json():
+    show_colors_json = {}
+    for show_color in Show_Color.query.all():
+        show_colors_json[show_color.show_colors_id] = {
+            'show_color': show_color.colors.color_id,
+            'brand_name': show_color.shows.brands.brand_name,
+            'color_id': show_color.color_id,
+            'color_name': show_color.colors.color_name,
+        }
+
+    return jsonify(show_colors_json)
+
+# <Show_Color show_colors_id=1 show_id=1 color_id=2
+
+
+@app.route('/_get_color_by_brand')
+def get_colors_by_brand_json():
+    brands = Brand.query.all()
+    brand_by_colors = {brand.brand_name: [] for brand in brands}
+    for brand in brands:
```

Comment:
if you're querying a table by primary key, you can just use `query.get`
http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.get